### ISSUE TITLE
[Bug] Higgs TTS: catastrophic WER outliers at large CUDA graph batch sizes (bs ≥ 16)

### ISSUE THREAD

**Issue by Ccyest** [sgl-project/sglang-omni#578]
## Problem

When running Higgs TTS with CUDA graph enabled at high concurrency (c=16 / c=32), a small fraction of samples (~0.1–0.4%) produce catastrophic WER (>50% per sample). These outliers drag corpus-level EN WER from ~1.3% to ~4.7%.

PR #503 identified the root cause for the *majority* of catastrophic samples: `pool[row_indices]` gather/scatter inside the captured graph produced wrong logits at certain batch positions for bs ≥ 8. The shadow-buffer fix (`_cg_active_*` pattern) reduced the catastrophic rate by 10×+, but did not eliminate it entirely.

PR #534 benchmark confirmed the residual tail:
- EN corpus WER = 4.68% (raw), 1.36% (excluding >50% outliers)
- 2 out of 1088 EN samples had >50% sample-level WER

The bug is specific to large CUDA graph batch size buckets (bs ≥ 16). At bs=1 the same prompts produce correct output.

## Reproduction

```bash
# Run SeedTTS EN full set at c=16, CUDA graph enabled
python benchmarks/eval/benchmark_tts_seedtts.py \
    --config examples/configs/higgs_tts.yaml \
    --lang en --concurrency 16 --num-samples 1088
```

Compare the per-sample WER distribution — catastrophic outliers (>50% WER) should appear at c≥16 but not at c=1.

## Root cause hypothesis

Something in the CUDA-graph-captured `decode_codebooks_batch_cg` path produces wrong logits or sampler state at specific batch positions when the graph is replayed at bs ≥ 16. The shadow-buffer pattern fixed the gather/scatter UB but there may be a second source of position-dependent divergence — possibly in:
- `batched_step_direct` state update (delay_count / eoc_countdown race at padded positions)
- `_decode_step_embeds_cg` embedding lookup when `_cg_active_last_codes` carries stale data from a previous graph replay at a different bs
- Padding-row state leaking into real rows via the `_cg_row_indices` mapping

## Suggested investigation

Dump `input_embeds`, `hidden_states_last`, `logits_BNV` at each decode step for a catastrophic sample (at bs ≥ 16) and the same prompt at bs=1. Diff layer-by-layer to find the first divergence point. The `HIGGS_DUMP_DIR` instrumentation mentioned in PR #503 was planned for this but not yet implemented.

## References

- PR #503 — CUDA Graph capture, shadow-buffer fix that reduced catastrophic rate 10×
- PR #534 — SeedTTS benchmark confirming residual outliers (2/1088 EN samples)
- Affected code: `sglang_omni/models/higgs_tts/model.py`, `decode_codebooks_batch_cg`
- Tracking issue: #478
